resources/definitions/Ender3_MicroSwiss_DirectDrive.def.json

```diff
@@ -0,0 +1,118 @@
+{
+    "name": "Ender-3 MicroSwiss DirectDrive",
+    "version": 2,
+    "inherits": "creality_ender3",
+    "metadata": {
+        "quality_definition": "creality_base",
+        "visible": true,
+        "platform": "creality_ender3.3mf",
+		"author": "GitHub.com/RFBomb",
+		"manufacturer": "Custom",
+
+		"machine_extruder_trains": {
+            "0": "MicroSwiss_DirectDrive_extruder_0",
+			"1": "MicroSwiss_DirectDrive_extruder_1"
+        }
+    },
+	
+    "overrides": {
+        "machine_name": { "default_value": "Ender-3 MicroSwiss DirectDrive" },
+		
+		"machine_extruder_count": { "default_value": 1, "maximum_value": "2" },
+		"extruders_enabled_count": { "default_value": 1 },
+        
+        "machine_use_extruder_offset_to_offset_coords": { 
+            "description_": "Apply the extruder offset to the coordinate system.",
+            "default_value": false 
+        },
+        
+        "machine_width": { "default_value": 230 },
+        "machine_depth": { "default_value": 230 },
+        "machine_height": { "default_value": 175 },
+		
+        "machine_disallowed_areas": {
+            "description": "A list of polygons with areas the print head is not allowed to enter.",
+			"default_value": [
+              [[-117.5, 117.5], [-117.5, 108], [117.5, 108], [117.5, 117.5]],
+              [[-117.5, -108], [-117.5, -117.5], [117.5, -117.5], [117.5, -108]]
+        ]},
+        "machine_head_with_fans_polygon": { "default_value": [
+                [-26, 34],
+                [-26, -32],
+                [32, -32],
+                [32, 34]
+            ]
+        },
+		"gantry_height": { 
+			"description": "The height difference between the tip of the nozzle and the gantry system (X and Y axes).",
+			"value": 35 
+		},
+        
+        "_Cura_Keywords":  "http://files.fieldofview.com/cura/Replacement_Patterns.html" ,
+        
+        "_Retraction_Comment": "MicroSwiss recomends maximum of 1.5mm @ 35mm/s for retraction settings when using their all-metal hot end with Direct Drive extruder",
+		"retraction_amount": { "default_value": 0.75 },
+		"retraction_extrusion_window": { "default_value":  "retraction_amount * 2" },
+        "retraction_speed": { "default_value": 15.0 }, 
+        
+
+        "machine_heat_zone_length":  { 
+            "description__": "The distance from the tip of the nozzle in which heat from the nozzle is transferred to the filament.",
+            "default_value": 16.0 
+        },
+        
+        "_ExtruderSwitchingSettings_Comment": "Settings to retract material when switching over to the other nozzle",
```

Comment:
There's a bunch of these that are not read by Cura. Could you remove them?

Reply:
They aren't read by cura, so shouldn't be a problem. (I formattted them purposefully to make it dissimilar to standard scheme). I found a few threads on how to use GitIgnore to ignore specifics lines, so I'll get that set up so these lines are no longer committed. They were there for my/anyone else that might modify them organization purposes mostly anyway.

Reply:
OK so after some test commits to my fork,  I found these items in the build test log:
`[MainThread] UM.Settings.DefinitionContainer._preprocessParsedJson [299]: Unable to override setting _Comment_Cura_Keywords` -> Apparently the test doesn't like them, but (other than writing to the log?) Cura doesn't seem to fail to use the files. As I said above, I'll remove them.

I also saw this : 
```
assert is_valid  # The definition has invalid JSON structure.
E           assert False
```
in reference to my file `Ender3_MicroSwiss_DirectDrive.def.json`. I take it this means that the structure is invalid? I'm a bit unsure why at the moment if that is the case. Other than a few extra spaces in a spot or two, the structure appears valid to me...

Reply:
Complete.